config.go

```diff
@@ -323,6 +325,37 @@ func golang(c *Config) {
 	}
 }
 
+// java config.
+func java(c *Config) {
+	if c.Hooks.Build.IsEmpty() {
+		// attempt to handle either Gradle or Maven builds and their respective wrappers
+		// assumes build/package results in shaded jar named server.jar
+		if util.Exists("pom.xml") {
+			if util.Exists("mvnw") {
+				c.Hooks.Build = config.Hook{`./mvnw clean package && cp target/server.jar .`}
+			} else {
+				c.Hooks.Build = config.Hook{`mvn clean package && cp target/server.jar .`}	
+			}				
+		} else if util.Exists("build.gradle") {
+			if util.Exists("gradlew") {
+				c.Hooks.Build = config.Hook{`./gradlew clean build && cp build/libs/server.jar .`}
+			} else {
+				c.Hooks.Build = config.Hook{`gradle clean build && cp build/libs/server.jar .`}	
+			}			
+		} else {
+			return
+		}
+	}
+
+	if c.Hooks.Clean.IsEmpty() {
+		c.Hooks.Clean = config.Hook{`rm server.jar`}
+	}
+
+	if c.Proxy.Command == "" && util.Exists("server.jar") {
```

Comment:
Hmm, I think this won't work.  `server.jar` will not exist at the point this hook is set. I think you can just remove that check entirely.

Reply:
Yeah, I realized this later and removed the check for `server.jar`. I'll push this change shortly.